--- configure	2019-01-07 16:37:04.774155801 +0300
+++ configure.new	2019-01-07 16:35:27.583620708 +0300
@@ -704,11 +704,14 @@
 	echo "MinGW32 C++ compiler: $MINGWCXX $MINGWCXXFLAGS"
 	echo "MinGW32 linker: $MINGWLD $MINGWLDFLAGS"
 	printf "Creating $CDIR/MuleToolBootstrapFiles/muletool-mingw32.exe.."
-	if $MINGWLD $MINGWLDFLAGS $MINGWCXXFLAGS -I$SOURCEDIR/src/ideintegration/muletool -DTOOLPREFIX=\"$PREFIX\" -DTOOLTARGET=\"$TARGETNAME\" -o "$CDIR/MuleToolBootstrapFiles/muletool-mingw32.exe" "$SOURCEDIR/src/ideintegration/muletool/muletoolclass.cpp" "$SOURCEDIR/src/ideintegration/muletool/main.cpp" ; then
+	if $MINGWLD $MINGWLDFLAGS -I$SOURCEDIR/src/ideintegration/muletool -DTOOLPREFIX=\"$PREFIX\" -DTOOLTARGET=\"$TARGETNAME\" -o "$CDIR/MuleToolBootstrapFiles/muletool-mingw32.exe" "$SOURCEDIR/src/ideintegration/muletool/muletoolclass.cpp" "$SOURCEDIR/src/ideintegration/muletool/main.cpp" ; then
 		echo ". done!"
 		HOSTCC="$MINGWCC"
 		HOSTCXX="$MINGWCXX"
 		HOSTLD="$MINGWLD"
+		HOSTCFLAGS="$MINGWCFLAGS"
+		HOSTCXXFLAGS="$MINGWCXXFLAGS"
+		HOSTLDFLAGS="$MINGWLDFLAGS"
 	else
 		echo ". failed!"
 		showerror 19 "Failed to link $CDIR/MuleToolBootstrapFiles/muletool-mingw32.exe"
